hal/armv7a/zynq7000/hal.c

```diff
@@ -86,26 +95,87 @@ addr_t hal_kernelGetAddress(addr_t addr)
 }
 
 
-int hal_memAddMap(addr_t start, addr_t end, u32 attr, u32 mapId)
+void hal_kernelEntryPoint(addr_t addr)
 {
-	return EOK;
+	hal_common.entry = addr;
 }
 
 
-void hal_cpuJump(addr_t addr)
+int hal_memoryAddMap(addr_t start, addr_t end, u32 attr, u32 mapId)
 {
-	syspage_save();
+	return 0;
+}
+
 
-	/* Tidy up */
-	hal_done();
+static void hal_getMinOverlappedRange(addr_t start, addr_t end, mapent_t *entry, mapent_t *minEntry)
+{
+	if ((start < entry->end) && (end > entry->start)) {
+		if (start > entry->start)
+			entry->start = start;
+
+		if (end < entry->end)
+			entry->end = end;
+
+		if (entry->start < minEntry->start) {
+			minEntry->start = entry->start;
+			minEntry->end = entry->end;
+			minEntry->type = entry->type;
+		}
+	}
+}
+
+
+int hal_memoryGetNextEntry(addr_t start, addr_t end, mapent_t *entry)
+{
+	mapent_t tempEntry = { 0 };
+	mapent_t minEntry = { .start = (addr_t)-1, .end = 0 };
+
+	if (start == end)
+		return -1;
+
+	/* plo: .bss */
+	tempEntry.start = (addr_t)_plo_bss;
+	tempEntry.end = (addr_t)_end;
+	tempEntry.type = hal_entryTemp;
+	hal_getMinOverlappedRange(start, end, &tempEntry, &minEntry);
+
+	/* syspage */
+	tempEntry.start = (addr_t)hal_common.hs;
+	tempEntry.end = tempEntry.start + SIZE_SYSPAGE;
+	tempEntry.type = hal_entryReserved;
+	hal_getMinOverlappedRange(start, end, &tempEntry, &minEntry);
+
+	/* plo: .stack */
+	tempEntry.start = (addr_t)ADDR_STACK;
+	tempEntry.end = tempEntry.start + SIZE_STACK;
+	tempEntry.type = hal_entryTemp;
+	hal_getMinOverlappedRange(start, end, &tempEntry, &minEntry);
+
+	if (minEntry.start != (addr_t)-1) {
+		entry->start = minEntry.start;
+		entry->end = minEntry.end;
+		entry->type = minEntry.type;
+
+		return 0;
+	}
+
+	return -1;
+}
```

Comment:
This code seems platfrom independent. Maybe we can think about sharing it between HALs in the future (something to keep in mind, no need to change this in this PR)

Reply:
I think that this code might not be applied to `ia32` because entries are obtained from BIOS. 
Nevertheless I will keep in mind to share this code between `ARM` platforms.